site/pages/create-consequence-operator/[disruptionId]/[consequenceIndex].page.tsx

```diff
@@ -51,6 +53,37 @@ const CreateConsequenceOperator = (props: CreateConsequenceOperatorProps): React
         queryParams["return"]?.includes(REVIEW_DISRUPTION_PAGE_PATH) ||
         queryParams["return"]?.includes(DISRUPTION_DETAIL_PAGE_PATH);
 
+    const [dataSource, setDataSource] = useState<Modes>(Modes.bods);
+
+    useEffect(() => {
+        const source = props.sessionWithOrg?.mode[pageState?.inputs?.vehicleMode as keyof ModeType];
+        if (source && dataSource !== source) {
+            setDataSource(source);
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [pageState?.inputs?.vehicleMode]);
+
+    const filterOperators = (operator: Operator) => {
+        const display =
+            !pageState.inputs.consequenceOperators?.find((selOp) => selOp.operatorNoc === operator.nocCode) &&
+            operator.dataSource === dataSource.toString();
+
+        if (pageState.inputs?.vehicleMode === "bus" && (operator.mode === "bus" || operator.mode === "")) {
+            return display;
+        } else if (
+            pageState.inputs?.vehicleMode === "tram" &&
+            (operator.mode === "tram" || operator.mode === "metro")
```

Comment:
does metro mode exist looking at vehicle modes there is none

Reply:
metro exists in the db which is mapped to tram

Reply:
could we use vehicle mode here?

Reply:
done